### PR TITLE
Issue 49987: Add labkey.experiment.lineage to Rlabkey

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
Add labkey.experiment.lineage API to support getting parent/child relationships for exp objects by LSID
https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=49987

#### Related PRs
- https://github.com/LabKey/testAutomation/pull/2012

#### Changes
- Add labkey.experiment.lineage function and doc info with example

### CODE REVIEW

File: Rlabkey/R/labkey.experiment.R [hunk 1]
```diff
@@ -198,3 +198,43 @@ labkey.experiment.saveRuns <- function(baseUrl=NULL, folderPath, protocolName, r
...
+labkey.experiment.lineage <- function(baseUrl=NULL, folderPath, lsids, options = NULL)
+{
+    baseUrl=labkey.getBaseUrl(baseUrl)
+
+    ## check required parameters
+    if (missing(folderPath))
+        stop (paste("A value must be specified for folderPath."))
+    if (missing(lsids))
+        stop (paste("A value must be specified for lsids."))
+    if (!missing(lsids) & !(is.vector(lsids) && is.atomic(lsids)))
```
Comment (labkey-klum) [inline Rlabkey/R/labkey.experiment.R:224]: interesting, I didn't know what an `atomic` vector was.

File: Rlabkey/man/labkey.experiment.lineage.Rd [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+labkey.experiment.lineage(
+  baseUrl="http://labkey/",
+  folderPath = "/home/OriginalFolder",
+  lsids=c("urn:lsid:labkey.com:Sample.519.Blood:23", "urn:lsid:labkey.com:Sample.519.Blood:12"),
+  options=list(
+    parents=TRUE,
+    children=TRUE,
+    depth=10,
+    #expType="Material",
```
Comment (labkey-klum) [inline Rlabkey/man/labkey.experiment.lineage.Rd]: Did you intend to comment this out?

**Review by labkey-klum — APPROVED** [review]

File: Rlabkey/R/labkey.experiment.R (hunk 1 above)
Comment (cnathe) [reply Rlabkey/R/labkey.experiment.R:224]: this was the suggestion when I googled how to check for a param as a vector. Apparently an R list will return is.vector() true for some reason

File: Rlabkey/man/labkey.experiment.lineage.Rd (hunk 2 above)
Comment (cnathe) [reply Rlabkey/man/labkey.experiment.lineage.Rd]: I was leaving it in as an example of another option, but I suppose it doesn't hurt to uncomment it. Done

**Comment by cnathe** [comment]
The CRAN package submission page is unavailable until Aug 16, so I guess I'll wait until then to finish this merge.

**Pull request merged**